src/main/scala/com/actian/spark_vector/vector/VectorJDBC.scala

```diff
@@ -29,9 +29,23 @@ import resource.managed
 
 /** Iterator over an ResultSet */
 abstract class ResultSetIterator[T](result: ResultSet) extends Iterator[T] {
+  private var gotNext = false
+  private var finished = false
+
   protected def extractor: ResultSet => T
-  override def hasNext: Boolean = result.next()
-  override def next(): T = extractor(result)
+
+  override def hasNext: Boolean = {
+    if (!gotNext && !finished) {
```

Comment:
`finished = !gotNext`  (`!gotNext` is already true from the above if) :8ball: 

...I'm not sure whre do you want to use this `finished` variable though